lib/ex_aws/sqs/client.ex

```diff
@@ -0,0 +1,122 @@
+defmodule ExAws.SQS.Client do
+  use Behaviour
+
+  @moduledoc """
+  The purpose of this module is to surface the ExAws.SQS API tied to a single
+  configuration chose, such that it does not need passed in with every request
+
+  Usage:
+  ```elixir
+  defmodule MyApp.SQS do
+    use ExAws.SQS.Client, otp_app: :my_otp_app
+  end
+  ```
+
+  In your config
+  ```elixir
+  config :my_otp_app, ExAws,
+    sqs: [], #SQS config goes here
+  ```
+  """
+
+  @type sqs_permission :: 
+    :all | 
+    :receive_message | 
+    :delete_message |
+    :change_message_visibility | 
+    :get_queue_attributes 
+  @type sqs_acl :: [{binary, sqs_permission}]
+  @type sqs_message_attribute_name ::
+    :all |
+    :sender_id | 
+    :sent_timestamp |
+    :approximate_receive_count |
+    :approximate_first_receive_timestamp |
+    :wait_time_seconds |
+    :receive_message_wait_time_seconds 
+  @type sqs_queue_attribute_name ::
+    :all | 
+    :approximate_number_of_messages |
+    :approximate_number_of_messages_not_visible | 
+    :visibility_timeout |
+    :created_timestamp | 
+    :last_modified_timestamp | 
+    :policy  
+  @type sqs_message :: [
+    {:message_body, binary},
+    {:delay_seconds, 0..900 | none},
+    {:attributes, Keyword.t }
+  ]
```

Comment:
I would say overall I want to have typespec that allows all things that the AWS service accepts, and rely on AWS to return validation errors rather than have typespec that doesn't allow something that AWS regards as valid. It's going to be a challenge to write typespec that lines up precisely with AWS, so I'm gonna err on the side of leniency in typespec so long as it at least covers all the options.